main/actions/src/main/scala/sbt/Tests.scala

```diff
@@ -273,7 +273,19 @@ object Tests {
   def discover(frameworks: Seq[Framework], analysis: CompileAnalysis, log: Logger): (Seq[TestDefinition], Set[String]) =
     discover(frameworks flatMap TestFramework.getFingerprints, allDefs(analysis), log)
 
-  def allDefs(analysis: CompileAnalysis) = analysis match { case analysis: Analysis => analysis.apis.internal.values.flatMap(_.api.definitions).toSeq }
+  def allDefs(analysis: CompileAnalysis) = analysis match {
```

Comment:
Everywhere in `PluginDiscovery` different methods explicitly talk about top-level modules. Do you have a sense how plugins declared in another object are discovered?